tracer/src/Datadog.Trace/Iast/EvidenceDictionaryConverter.cs

```diff
@@ -0,0 +1,167 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System.Collections.Generic;
+
+namespace Datadog.Trace.Iast;
+
+/// <summary>
+/// Serialize <see cref="Datadog.Trace.Iast.Evidence"/> struct to a dictionary
+/// </summary>
+internal class EvidenceDictionaryConverter
+{
+    // When not redacted output is:
+    // "valueParts": [
+    //   { "value": "SELECT * FROM Users WHERE " },     -> Not tainted part
+    //   { "source": 0, "value": "Name='name'" }        -> Tainted part from source 0
+    //   { "value": " and RoleId=25" }                  -> Not tainted part
+    // ]
+    //
+    // When redacted output is:
+    // "valueParts": [
+    //   { "value": "SELECT * FROM Users WHERE " },            -> Not tainted part
+    //   { "source": 0, "value": "Name='" },                   -> Tainted part from source 0
+    //   { "source": 0, "redacted": true, "pattern": "abcd" }, -> Redacted tainted part from source 0
+    //   { "source": 0, "value": "'" },                        -> Tainted part from source 0
+    //   { "value": " and RoleId=" },                          -> Not tainted part
+    //   { "redacted": true }                                  -> Redacted not tainted part
+    // ]
+
+    // Explanation:
+    //   Input is a string value, tainted ranges and sensitive ranges.
+    //   We must produce an array of "value parts", with are segments of the input value.
+    //   Non sensitive parts are plain text. Sensitive ranges must be redacted.
+    //   If a tainted range intersects with a sensitive range, the corresponding source must be redacted also.
+
+    private bool _redactionEnabled;
+    private int _maxValueLength;
+
+    public EvidenceDictionaryConverter(int maxValueLength, bool redactionEnabled)
+    {
+        _redactionEnabled = redactionEnabled;
+        _maxValueLength = maxValueLength;
+    }
+
+    public Dictionary<string, object> EvidenceToDictionary(Evidence evidence)
+    {
+        var result = new Dictionary<string, object>();
+
+        if (evidence.Ranges == null || evidence.Ranges.Length == 0)
+        {
+            TruncationUtils.InsertTruncatableValue(result, "value", evidence.Value, _maxValueLength);
+        }
+        else
+        {
+            var valueParts = new List<object>();
+            if (_redactionEnabled)
+            {
+                result["valueParts"] = ToRedactedDictionary(valueParts, evidence.Value!, evidence.Ranges, evidence.Sensitive);
+            }
+            else
+            {
+                result["valueParts"] = ToJsonTaintedDictionary(valueParts, evidence.Value!, evidence.Ranges);
+            }
+        }
+
+        return result;
+    }
+
+    private List<object> ToJsonTaintedDictionary(List<object> valueParts, string value, Range[] ranges)
```

Comment:
Isn't it a List?

Reply:
I changed it, yep it might be confusing, but I suffixed it with Dictionary because that was creating a list of dictionaries